### PR TITLE
[CIRCLE-35388] Add Windows Runner installation instructions

### PR DESCRIPTION
**Pull request by tiaandcoffee**
# Description
Added instructions for Windows Runner installation, per [this ticket](https://circleci.atlassian.net/browse/CIRCLE-35388?atlOrigin=eyJpIjoiYTIzNDE2MmFlMGI3NGUzNDliZjI3NTBhYmNmODQ5YTQiLCJwIjoiaiJ9) and, more specifically, [this doc](https://docs.google.com/document/d/1llXil_5deW0pa0DGPbwOlIUSmFITmvHjH7cQ7Aeq5R4/edit#heading=h.9venwp3q5wy6). Took some creative liberties to align with the existing installation instructions and flow of the page as a whole, definitely open to feedback.
Some additional small edits on the page aside from the instructions, such as moving the `Configuration file reference` section to go after all of the different platform-specific installation instructions.

# Reasons
Ticket: https://circleci.atlassian.net/browse/CIRCLE-35388?atlOrigin=eyJpIjoiYTIzNDE2MmFlMGI3NGUzNDliZjI3NTBhYmNmODQ5YTQiLCJwIjoiaiJ9

### CODE REVIEW

File: jekyll/_cci2/runner-installation.adoc [hunk 1]
```diff
@@ -365,6 +367,76 @@ sudo launchctl load '/Library/LaunchDaemons/com.circleci.runner.plist'
...
+. Run the following in your PowerShell:
++
+```
+Set-ExecutionPolicy Bypass -Scope Process -Force; [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor 3072; 
+./Install-CircleCIRunner.ps1
+```
++
+The installation will be output into your PowerShell interface.
+
+. As part of the installation, the configuration file for the Runner (`launch-agent-config.yaml`) will open in Notepad. Please fill the file out with the requested information (see <<Configuration file reference, Configuration file reference>>). The configuration file is located in the installation directory - `C:\Program Files\CircleCI`, by default.
```
Comment (MaT1g3R) [inline jekyll/_cci2/runner-installation.adoc:436]:
```suggestion
. As part of the installation, the configuration file for the Runner (`launch-agent-config.yaml`) will opened in Notepad. Please fill the file out with the requested information (see <<Configuration file reference, Configuration file reference>>). The configuration file is located in the installation directory - `C:\Program Files\CircleCI`, by default.
```

Comment (tiaandcoffee) [reply jekyll/_cci2/runner-installation.adoc:436]: Hey @MaT1g3R, thanks for the review :) I am open to hear your thoughts on the change you suggested, but I think the way I have it is correct? Or what about `be opened`?

Comment (MaT1g3R) [reply jekyll/_cci2/runner-installation.adoc:436]: whoops, I assumed there's a `be` there for some reason. You can ignore this.

Comment (tiaandcoffee) [reply jekyll/_cci2/runner-installation.adoc:436]: thank you! anything else standing in the way of a stamp?

Comment (MaT1g3R) [reply jekyll/_cci2/runner-installation.adoc:436]: no, i think it's good!

**Review by MaT1g3R — APPROVED** [review]

**Review by stalmy — APPROVED** [review]

**Review by stalmy — APPROVED** [review]

File: jekyll/_cci2/runner-installation.adoc [hunk 2]
```diff
@@ -365,6 +367,76 @@ sudo launchctl load '/Library/LaunchDaemons/com.circleci.runner.plist'
...
+
+=== Build the Docker image
+
+```bash
+docker build --file ./Dockerfile.runner.extended .
+```
+
+=== Start the Docker container
+
+NOTE: The environment variable values are not to the `docker` command, so these environment variables are not visible in `ps` output.
```
Comment (teesloane) [inline jekyll/_cci2/runner-installation.adoc:397]: > "The environment variable values are not... visible? available? ... to the `docker` ...

Are you missing a word here?

File: jekyll/_cci2/runner-installation.adoc [hunk 3]
```diff
@@ -365,6 +367,76 @@ sudo launchctl load '/Library/LaunchDaemons/com.circleci.runner.plist'
...
+
+== Windows Installation
+
+NOTE: This has been tested for Windows Server 2019 and Windows Server 2016, both in Datacenter Edition with Desktop Experience.
+
+With this procedure, you install CircleCI Launch Agent and its dependencies, i.e., Chocolatey, Git and Gzip, on your Windows Server.
+
+Since this setup creates a new local administrator user that runs CircleCI jobs, your Windows Server needs to be capable of creating local users and accepting remote logons for them.
+
+* The Windows Server needs to be a domain member when you set this up in domain environments.The runner instance cannot run on a Windows Server operating as a domain controller.
```
Comment (teesloane) [inline jekyll/_cci2/runner-installation.adoc:419]:
```suggestion
* The Windows Server needs to be a domain member when you set this up in domain environments. The runner instance cannot run on a Windows Server operating as a domain controller.
```

**Review by teesloane — CHANGES_REQUESTED** [review]
Two small nits/questions! Otherwise, this looks great!

**Review by teesloane — APPROVED** [review]

**Pull request merged**